frontend/src/framework/HoverService.ts

```diff
@@ -0,0 +1,195 @@
+    [HoverTopic.REALIZATION]: number | null;
+    [HoverTopic.TIMESTAMP]: number | null;
+    [HoverTopic.ZONE]: string | null;
+    [HoverTopic.REGION]: string | null;
+    [HoverTopic.FACIES]: string | null;
+    [HoverTopic.WORLD_POS]: { x?: number; y?: number; z?: number } | null;
+};
+
+type ThrottledPublishFunc = _.DebouncedFunc<<T extends keyof HoverData>(topic: T, newValue: HoverData[T]) => void>;
+
+/**
+ * Service to manage hover data across the application. Contains two different data sets, one that is updated immediately, and one that is updated at a throttled interval.
+ */
+export class HoverService {
+    // Currently available hover-data. The two objects are updated at different rates,
+    // but will contain the same data when the system is not actively hovering
+    private _hoverData: Partial<HoverData> = {};
+    private _throttledHoverData: Partial<HoverData> = {};
+    private _lastHoveredModule: string | null = null;
+
+    // Throttling. Each topic is updated with its own throttle method.
+    private _topicThrottleMap = new Map<keyof HoverData, ThrottledPublishFunc>();
+    private _dataUpdateThrottleMs = HOVER_UPDATE_THROTTLE_MS;
+
+    // Delegate to handle update notifications
+    private _publishSubscribeDelegate = new PublishSubscribeDelegate<HoverData>();
+
+    private _getOrCreateTopicThrottleMethod(topic: keyof HoverData): ThrottledPublishFunc {
+        if (!this._topicThrottleMap.has(topic)) {
+            const throttledMethod = throttle(this._doThrottledHoverDataUpdate.bind(this), this._dataUpdateThrottleMs, {
+                // These settings make it so notifications are only pushed *after* the throttle timer elapses
+                leading: false,
+                trailing: true,
+            });
+
+            this._topicThrottleMap.set(topic, throttledMethod);
+        }
+
+        // If-block above gurantees this is non-null
+        return this._topicThrottleMap.get(topic)!;
+    }
+
+    private _doThrottledHoverDataUpdate<T extends keyof HoverData>(topic: T, value: HoverData[T]): void {
+        this._throttledHoverData[topic] = value;
+        this.getPublishSubscribeDelegate().notifySubscribers(topic);
+    }
+
+    /**
+     * @returns The publish-subscribe delegate used by this service
+     */
+    getPublishSubscribeDelegate(): PublishSubscribeDelegate<HoverData> {
+        return this._publishSubscribeDelegate;
+    }
+
+    /**
+     * Gets the current value for a given topic. If the requesting module is the one currently hovering, it will get the latest data immediately, otherwise it will get the latest throttled data.
+     * @param topic The HoverTopic to get the value for
+     * @param moduleInstanceId The id of the module requesting the value.
+     * @returns The latest or throttled value for the given topic
+     */
+    getTopicValue<T extends HoverTopic>(topic: T, moduleInstanceId: string): HoverData[T] | null {
+        // ? Should  this be an  opt-in functionality?
```

Comment:
Do you want to make it opt-in for the moduleInstance which is last hovered, or do you want it opt-in for modules even if they are not the last hovered?

I think current functionality seems like a good solution. We can rather make it opt-in when we see the need for it?